numpy/ctypeslib.py

```diff
@@ -466,53 +471,89 @@ def _ctype_from_dtype(dtype):
 
     def as_ctypes_type(dtype):
         r"""
-        Convert a dtype into a ctypes type.
-
-        Parameters
-        ----------
-        dtype : dtype
-            The dtype to convert
-
-        Returns
-        -------
-        ctype
-            A ctype scalar, union, array, or struct
-
-        Raises
-        ------
-        NotImplementedError
-            If the conversion is not possible
-
-        Notes
-        -----
-        This function does not losslessly round-trip in either direction.
-
-        ``np.dtype(as_ctypes_type(dt))`` will:
-
-         - insert padding fields
-         - reorder fields to be sorted by offset
-         - discard field titles
-
-        ``as_ctypes_type(np.dtype(ctype))`` will:
-
-         - discard the class names of `ctypes.Structure`\ s and
-           `ctypes.Union`\ s
-         - convert single-element `ctypes.Union`\ s into single-element
-           `ctypes.Structure`\ s
-         - insert padding fields
+            as_ctypes_type(dtype)
+
+            Convert a dtype into a ctypes type.
+
+            Parameters
+            ----------
+            dtype : dtype
+                The dtype to convert
+
+            Returns
+            -------
+            ctype
+                A ctype scalar, union, array, or struct
+
+            Raises
+            ------
+            NotImplementedError
+                If the conversion is not possible
+
+            Notes
+            -----
+            This function does not losslessly round-trip in either direction.
+            ``np.dtype(as_ctypes_type(dt))`` will:
+              - insert padding fields
+              - reorder fields to be sorted by offset
+              - discard field titles
+            ``as_ctypes_type(np.dtype(ctype))`` will:
+              - discard the class names of `ctypes.Structure`\ s and
+                `ctypes.Union`\ s
+              - convert single-element `ctypes.Union`\ s into single-element
+                `ctypes.Structure`\ s
+              - insert padding fields
+
+            Examples
+            --------
+            >>> np.ctypeslib.as_ctypes_type(np.int32)
+            ctypes.c_int
+
+            >>> np.ctypeslib.as_ctypes_type(np.float32)
+            ctypes.c_float
 
         """
         return _ctype_from_dtype(_dtype(dtype))
 
 
     def as_array(obj, shape=None):
         """
-        Create a numpy array from a ctypes array or POINTER.
-
-        The numpy array shares the memory with the ctypes object.
+            as_array(obj, shape=None)
```

Comment:
Hmmm, maybe this is not actually bad.  I don't see it in the rendered documentation (you go to `ci/circleci: build artifact — Link to 0/doc/build/html/index.html` in the Build matrix at the bottom and click on "Details" then the new documentation opens up.)